utils/getUniqueCategories.ts

```diff
@@ -1,19 +1,21 @@
-import Expenses from '../config/finance/json-data/2024/Expenses.json';
-
 /**
  * Retrieves unique expense categories from the Expenses data.
  *
  * @param {Object} expenses - The expenses data.
  * @returns {string[]} An array of unique expense categories.
  */
-export const getUniqueCategories = (): string[] => {
+export const getUniqueCategories = async (): Promise<string[]> => {
+    const currentYear = new Date().getFullYear();
     const allCategories: string[] = [];
+    const Expenses = (await import(`../config/finance/json-data/${currentYear}/Expenses.json`)).default;
```

Comment:
Instead of providing imports inside code, can't we lazy import these components/files?

https://nextjs.org/docs/app/building-your-application/optimizing/lazy-loading#nextdynamic

Reply:
We are importing the json files normally now!